File: mvvm-java/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/search/SearchBindingAdapters.java
package com.wasisto.githubuserfinder.android.ui.search;

import androidx.databinding.BindingAdapter;
import androidx.recyclerview.widget.RecyclerView;

import com.wasisto.githubuserfinder.domain.models.SearchHistoryItem;
import com.wasisto.githubuserfinder.domain.models.SearchUsersResult;

import java.util.List;

public class SearchBindingAdapters {

    @BindingAdapter({"searchResultItems", "viewModel"})
    public static void searchResultItems(RecyclerView recyclerView, List<SearchUsersResult.Item> searchResultItems,
                                         SearchViewModel viewModel) {
        if (searchResultItems != null) {
            SearchResultAdapter searchResultAdapter = (SearchResultAdapter) recyclerView.getAdapter();
            if (searchResultAdapter == null) {
                recyclerView.setAdapter(
                        new SearchResultAdapter(viewModel) {{
                            setData(searchResultItems);
                        }}
                );
            }
        }
    }

    @BindingAdapter({"searchHistoryItems", "viewModel"})
    public static void searchHistoryItems(RecyclerView recyclerView, List<SearchHistoryItem> searchHistoryItems,
                                          SearchViewModel viewModel) {
        if (searchHistoryItems != null) {
            SearchHistoryAdapter searchHistoryAdapter = (SearchHistoryAdapter) recyclerView.getAdapter();
            if (searchHistoryAdapter == null) {
                recyclerView.setAdapter(
                        new SearchHistoryAdapter(viewModel) {{
                            setData(searchHistoryItems);
                        }}
                );
            }
        }
    }
}

File: mvvm-kotlin/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/search/SearchBindingAdapters.kt
package com.wasisto.githubuserfinder.android.ui.search

import androidx.databinding.BindingAdapter
import androidx.recyclerview.widget.RecyclerView
import com.wasisto.githubuserfinder.domain.models.SearchHistoryItem
import com.wasisto.githubuserfinder.domain.models.SearchUsersResult

@BindingAdapter("searchResultItems", "viewModel")
fun searchResultItems(recyclerView: RecyclerView, searchResultItems: List<SearchUsersResult.Item>?,
                      viewModel: SearchViewModel) {
    searchResultItems?.let {
        recyclerView.adapter = (recyclerView.adapter as? SearchResultAdapter ?: SearchResultAdapter(viewModel)).apply {
            data = searchResultItems
        }
    }
}

@BindingAdapter("searchHistoryItems", "viewModel")
fun searchHistoryItems(recyclerView: RecyclerView, searchHistoryItems: List<SearchHistoryItem>?,
                       viewModel: SearchViewModel) {
    searchHistoryItems?.let {
        recyclerView.adapter = (recyclerView.adapter as? SearchHistoryAdapter ?: SearchHistoryAdapter(viewModel)).apply {
            data = searchHistoryItems
        }
    }
}